gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -189,10 +192,23 @@ private void onShowDialog(
     }
   }
 
-  private void observeSelectPhotoClicks(PhotoFieldViewModel viewModel) {
-    viewModel
+  private void observeSelectPhotoClicks(PhotoFieldViewModel fieldViewModel) {
+    fieldViewModel
         .getShowDialogClicks()
-        .observe(this, __ -> onShowPhotoSelectorDialog(viewModel.getField()));
+        .observe(this, __ -> onShowPhotoSelectorDialog(fieldViewModel.getField()));
+  }
+
+  private void observePhotoAdded(PhotoFieldViewModel fieldViewModel) {
+    viewModel
+        .getPhotoUpdated()
+        .observe(
+            this,
+            map -> {
+              Entry<Field, String> entry = map.entrySet().iterator().next();
+              if (entry.getKey().equals(fieldViewModel.getField())) {
```

Comment:
Why iterate over keys here instead of just calling `map.get()`?

Reply:
Thanks for noticing. Done!